examples/event-listener.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace event_listener;
+
+use Castor\Attribute\AsListener;
+use Castor\Attribute\AsTask;
+use Castor\Event\BeforeExecuteTaskEvent;
+
+use function Castor\dispatch_event;
+use function Castor\io;
+
+#[AsTask(description: 'Task that need ensure the environment is set')]
+function my_task(): void
+{
+    io()->writeln('Hello from task!');
+}
+
+// Listen to a provided event
+#[AsListener(event: BeforeExecuteTaskEvent::class, priority: 1)]
+function my_listener(BeforeExecuteTaskEvent $event): void
+{
+    $taskName = $event->getCommand()->getName();
```

Comment:
```suggestion
    $taskName = $event->getTask()->getName();
```
Would feel more consistent, since it's a BeforeExecuteTaskEvent

Reply:
I named it because the main object is named `TaskCommand` and inherit of Symfony `Command` class